src/Dropdown/index.jsx

```diff
@@ -1,225 +1,198 @@
 import React from 'react';
-import classNames from 'classnames';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 
 import Button from '../Button';
 
-export const triggerKeys = {
-  OPEN_MENU: ['ArrowDown'],
-  CLOSE_MENU: ['Escape'],
-  NAVIGATE_DOWN: ['ArrowDown', 'Tab'],
-  NAVIGATE_UP: ['ArrowUp'],
-  SELECT_MENU_ITEM: ['Enter', ' '],
-};
-
 class Dropdown extends React.Component {
-  static isTriggerKey(action, keyName) {
-    return triggerKeys[action].indexOf(keyName) > -1;
-  }
-
   constructor(props) {
     super(props);
-
-    this.addEvents = this.addEvents.bind(this);
-    this.handleDocumentClick = this.handleDocumentClick.bind(this);
-    this.handleToggleKeyDown = this.handleToggleKeyDown.bind(this);
-    this.handleMenuKeyDown = this.handleMenuKeyDown.bind(this);
-    this.removeEvents = this.removeEvents.bind(this);
-    this.toggle = this.toggle.bind(this);
-
-    this.menuItems = [];
     this.state = {
       open: false,
-      focusIndex: 0,
     };
+    this.menuItems = React.createRef();
   }
 
-  componentWillUpdate(_, nextState) {
-    if (nextState.open) {
-      this.addEvents();
-    } else {
-      this.removeEvents();
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.open !== this.state.open) {
+      if (this.state.open) {
+        this.focusFirst();
+      } else {
+        this.toggleButton.focus();
+      }
     }
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (this.state.open) {
-      this.menuItems[this.state.focusIndex].focus();
-    } else if (prevState.open && this.toggleElem) {
-      this.toggleElem.focus();
-    }
+  componentWillUnmount = () => {
```

Comment:
This doesn't need to be an arrow function right?

Reply:
Correct, updated open() and close() as well.